src/common/metadata.c

```diff
@@ -117,6 +118,31 @@ const char *dt_metadata_get_key(const uint32_t keyid)
     return NULL;
 }
 
+const char *dt_metadata_get_subkey(const uint32_t keyid)
+{
+  if(keyid < DT_METADATA_NUMBER)
+  {
+    char *t = g_strrstr(dt_metadata_def[keyid].key, ".");
+    return t + 1;
+  }
+  else
+    return NULL;
+}
+
+const char *dt_metadata_get_key_by_subkey(const char *subkey)
+{
+  if(subkey)
+  {
+    for(unsigned int i = 0; i < DT_METADATA_NUMBER; i++)
+    {
+      char *t = g_strrstr(dt_metadata_def[i].key, ".");
+      if(!g_strcmp0(t + 1, subkey))
```

Comment:
same here, we need to check for t == NULL

Reply:
sure